### DISCUSSION TITLE
unable to persist data on asyncstorage

### DISCUSSION THREAD

**Discussion by leamandeep** [pmndrs/zustand#1857] [Q&A]
Hi, I'm using `asyncstorage`.  I followed docs https://docs.pmnd.rs/zustand/integrations/persisting-store-data#simple-example. But I'm not able persist data

```jsx
import create from "zustand";
import { persist, createJSONStorage } from "zustand/middleware";
import AsyncStorage from "@react-native-async-storage/async-storage";

export const useBearStore = create(
  persist(
    (set) => ({
      user: {},
      setUser: (user) => set((state) => ({ user: user })),
      isLogin: false,
      setIsLogin: (isLogin) => set((state) => ({ isLogin: isLogin })),
}), {
      name: "user",
      partialize: (state) => ({ user: state.user }),
      storage: createJSONStorage(() => AsyncStorage),
    },
    {
      name: "isLogin",
      partialize: (state) => ({ isLogin: state.isLogin }),
      storage: createJSONStorage(() => AsyncStorage),
    }
  )
```

**Comment by dbritto-dev** [comment]
@elamandeep would you mind sharing us what error you are having? BTW,  would you mind trying without partializing `user` and `isLogin`?

**Reply by leamandeep** [reply]
I'm not getting any error. I stored values inside `user` and `isLogin`. But it's not persisting that value. 

I can't remove partialize because I have more property inside my store which I don't want to store

**Reply by leamandeep** [reply]
@dbritto-dev any help?

**Reply by dbritto-dev** [reply]
> @dbritto-dev any help?

I'll create a snack example for you

**Reply by leamandeep** [reply]
@dbritto-dev should i create snack?

**Reply by leamandeep** [reply]
@dbritto-dev I made a snack. But i'm getting error
https://snack.expo.dev/@elamandeep/disrespectful-pizza?platform=web

**Reply by dbritto-dev** [reply]
@elamandeep you need to specify the zustand version (4.x)

**Reply by leamandeep** [reply]
@dbritto-dev I have added zustand with a specific version. Still getting the same error. I don't know how to fix it `Unable to resolve module 'zustand/middleware.js'`. But have a glance at my code. I used that piece of code for storing data on async storage.

**Reply by leamandeep** [reply]
@dbritto-dev Here's the updated snack https://snack.expo.dev/@lucktale/disrespectful-pizza?platform=web
I'm sure this hydration issue

**Reply by dbritto-dev** [reply]
> @dbritto-dev Here's the updated snack https://snack.expo.dev/@lucktale/disrespectful-pizza?platform=web
> I'm sure this hydration issue

Thanks, I'll take a look of that
